### ISSUE TITLE
ANX determination

### ISSUE THREAD

**Issue by lmar76** [stcorp/procsim#24]
According to [CHANGELOG](https://github.com/stcorp/procsim/blob/7877ccdeffa6b40d14e2e6626fb656ab5e087eb6/CHANGELOG), with version 1.5.0, you have introduced the possibility to "Read orbit prediction files and retrieve ANX timestamps.".

I have modified the scenario configuration for the level-0 step 1 (i.e. raw slicing) by commenting the "anx" information:

```json
        {
            "name": "LOAD_TDS - SM idx 0 DTID 7217244 L0PFStep1_022",
            "file_name": "L0PFStep1",
            "processor_name": "L0PFStep1_022",
            "processor_version": "01.00",
            "task_name": "L0PFStep1",
            "task_version": "01.00",
            "log_level": "debug",
            // "anx": [
            //     "2017-03-02T06:05:51.514000Z",
            //     "2017-03-02T07:44:02.496000Z"
            // ],
            "begin_end_position_from_toi": true,
            "outputs": [
                {
                    "type": "RAWS022_10",
                    "metadata_source": ".*RAW_022_10_.*"
                }
            ]
        },
```

Then, I started the processing simulation but the following error is traced:

```
% /opt/pnode/bin/L0PFStep1 BIO_L0PFStep1_022_01.00_LOAD-TDS-DT7217244_20220112T231817_JobOrder.xml
2022-01-29T17:33:27.715435 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Procsim v1.5.0 processor stub simulator
2022-01-29T17:33:27.715489 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Simulate scenario LOAD_TDS - SM idx 0 DTID 7217244 L0PFStep1_022
2022-01-29T17:33:27.715500 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Read JobOrder BIO_L0PFStep1_022_01.00_LOAD-TDS-DT7217244_20220112T231817_JobOrder.xml
2022-01-29T17:33:27.715508 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [D] JobOrder validation against schema: OK
2022-01-29T17:33:27.715515 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Read task L0PFStep1 from the JobOrder
2022-01-29T17:33:27.715525 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Input type RAW_022_10: BIO_RAW_022_10_20170302T060707_20170302T060937_D20170302T073617_01_8YJQIX
2022-01-29T17:33:27.715533 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Input type AUX_PP0___: BIO_AUX_PP0____20170101T000000_20500101T000000_01_8VG000
2022-01-29T17:33:27.715547 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Input type AUX_ERP___: BIO_AUX_ERP____20170302T000000_20170501T000000_01_91M800
2022-01-29T17:33:27.715555 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Input type MPL_ORBPRE: BIO_TEST_MPL_ORBPRE_20170101T060001_20170910T063749_0001.EOF
2022-01-29T17:33:27.725849 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [E] ANX must be configured for RAWSxxx_10 product
2022-01-29T17:33:27.725875 ESR01104747 L0PFStep1_022 01.00 L0PFStep1 [0000002663]: [I] Terminate with code 1
```

Please note that, as indicated in the log, the predicted orbit file (containing the anx values) was read.

Is there anything wrong in the scenario configuration? How it is possible to use this new feature allowing to parse anx values directly from the predicted orbit file?

**Comment by mivermeu** [comment]
Thank you for bringing this to our attention. Although orbital prediction files were parsed, a check was still in place that required ANX information in the scenario. This has been resolved with PR #25. With this PR, ANX information is not mandatory in the scenario if an orbital prediction file is present. Do note that ANX information from the scenario takes precedence over orbital prediction files if both are supplied: the latter will not be read if the former is present.

**Comment by lmar76** [comment]
Successfully tested with v1.6.0.
